cmd/soroban-rpc/internal/db/ledger.go

```diff
@@ -75,6 +75,38 @@ func (r ledgerReader) GetLedger(ctx context.Context, sequence uint32) (xdr.Ledge
 
 // GetLedgerRange pulls the min/max ledger sequence numbers from the meta table.
 func (r ledgerReader) GetLedgerRange(ctx context.Context) (ledgerbucketwindow.LedgerRange, error) {
+	var latestLedgerSeqCache uint32
+	var latestLedgerCloseTimeCache int64
+
+	r.db.cache.RLock()
+	latestLedgerSeqCache = r.db.cache.latestLedgerSeq
+	latestLedgerCloseTimeCache = r.db.cache.latestLedgerCloseTime
+	r.db.cache.RUnlock()
+
+	// Make use of the cached latest ledger seq and close time to query only the oldest ledger details.
+	if latestLedgerSeqCache != 0 {
+		query := sq.Select("meta").
+			From(ledgerCloseMetaTableName).
+			Where(
+				fmt.Sprintf("sequence = (SELECT MIN(sequence) FROM %s)", ledgerCloseMetaTableName),
+			)
+		var lcm []xdr.LedgerCloseMeta
+		if err := r.db.Select(ctx, &lcm, query); err != nil {
+			return ledgerbucketwindow.LedgerRange{}, fmt.Errorf("couldn't query ledger range: %w", err)
+		}
+
+		return ledgerbucketwindow.LedgerRange{
+			FirstLedger: ledgerbucketwindow.LedgerInfo{
+				Sequence:  lcm[0].LedgerSequence(),
```

Comment:
you should add a check to assert lcm is not empty. I know that we don't expect it to be empty because of the cache indicates that ledger exists but it is still worth including the check imo